doc/source/user/build_system/core_files.rst

```diff
@@ -24,30 +26,44 @@ Things one should know about YAML
 
   .. code-block:: yaml
 
-     [ item1, item2 ]
+     [ 'item1', 'item2' ]
 
   is semantically identical to
 
   .. code-block:: yaml
 
-     - item1
-     - item2
+     - 'item1'
+     - 'item2'
 
   The same is true for dictionaries (key/value pairs).
 
   .. code-block:: yaml
 
-     { key1: value1, key2: value2 }
+     { key1: 'value1', key2: 'value2' }
 
   is semantically identical to
 
   .. code-block:: yaml
 
-     key1: value1
-     key2: value2
+     key1: 'value1'
+     key2: 'value2'
 
   In most cases, the longer (second) form is preferred, as it is easier to make changes while keeping the diff easy to read.
 
+* **We recommend always adding single quotes around strings used as value.**
+  In most cases, strings in YAML do not need to be surrounded by (single or double) quotation marks.
+  However, the exact rules when quoting is needed are not easily summarized without in-depth understanding of the YAML language syntax.
+  At minimum quotation marks are *required* when strings start with an exclamation mark.
```

Comment:
Double quotes would work almost equally well for our use case, and might be slightly more "typical" for strings. No strong feelings there, I went with single quotes as they are slightly "safer".

Reply:
Me neither. I tend to use double quotes myself but that doesn't mean anyone else has to follow that

Reply:
Yeah, my muscle memory also clearly prefers double quotes. I thought about it again and the benefits of using single quotes aren't really worth using the "less common" quotes for strings. I switched to double quotes instead.